addOns/client/src/main/java/org/zaproxy/addon/client/zest/ExtensionClientZest.java

```diff
@@ -0,0 +1,92 @@
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2023 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.client.zest;
+
+import java.util.List;
+import net.sf.json.JSONObject;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.Extension;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.zaproxy.addon.client.ExtensionClientIntegration;
+import org.zaproxy.zap.extension.zest.ExtensionZest;
+
+public class ExtensionClientZest extends ExtensionAdaptor {
+
+    public static final String NAME = "ExtensionAuthhelperAjax";
+
+    private ZestClientRecordingHelper clientHelper;
+
+    private static final List<Class<? extends Extension>> DEPENDENCIES =
+            List.of(ExtensionZest.class);
+
+    public ExtensionClientZest() {
+        super(NAME);
+    }
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+        ExtensionClientIntegration extClient =
+                Control.getSingleton()
+                        .getExtensionLoader()
+                        .getExtension(ExtensionClientIntegration.class);
+        clientHelper = new ZestClientRecordingHelper(this);
+        extClient.setClientRecorderHelper(clientHelper);
+    }
+
+    @Override
+    public boolean canUnload() {
+        return true;
+    }
+
+    @Override
+    public void unload() {
+        ExtensionClientIntegration extClient =
+                Control.getSingleton()
+                        .getExtensionLoader()
+                        .getExtension(ExtensionClientIntegration.class);
+        extClient.removeClientRecorderHelper();
+    }
+
+    @Override
+    public List<Class<? extends Extension>> getDependencies() {
+        return DEPENDENCIES;
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString("client.zest.desc");
+    }
+
+    @Override
+    public String getUIName() {
+        return Constant.messages.getString("client.zest.name");
+    }
+
+    void addZestStatement(JSONObject stmt) throws Exception {
+        ExtensionZest extZst =
+                Control.getSingleton().getExtensionLoader().getExtension(ExtensionZest.class);
+        if (extZst.isClientRecodingActive()) {
```

Comment:
IMO it shouldn't be responsibility of this extension to verify if the recording is active, that should be done by the Zest extension.